esmvalcore/preprocessor/_trend.py

```diff
@@ -0,0 +1,181 @@
+    if np.ma.is_masked(y_arr):
+        x_arr = x_arr[~y_arr.mask]
+        y_arr = y_arr[~y_arr.mask]
+    return (x_arr, y_arr)
+
+
+def _slope(x_arr, y_arr):
+    """Calculate slope."""
+    x_demean = x_arr - x_arr.mean()
+    y_demean = y_arr - y_arr.mean()
+    slope = (x_demean * y_demean).sum() / (x_demean**2).sum()
+    return slope
+
+
+def _get_slope(y_arr, x_arr):
+    """Calculate slope between X and Y array."""
+    (x_arr, y_arr) = _mask_xy(x_arr, y_arr)
+
+    # If less than 2 points, slope calculation not possible
+    if len(y_arr) < 2:
+        return np.nan
+
+    # Calculate slope
+    return _slope(x_arr, y_arr)
+
+
+def _get_slope_stderr(y_arr, x_arr):
+    """Calculate standard error of slope between X and Y array."""
+    (x_arr, y_arr) = _mask_xy(x_arr, y_arr)
+
+    # If less than 2 points, slope stderr calculation not possible
+    if len(y_arr) < 2:
+        return np.nan
+
+    # Calculate standard error of slope
+    if len(y_arr) == 2:
+        return 0.0
+    dof = len(y_arr) - 2
+    x_mean = x_arr.mean()
+    y_mean = y_arr.mean()
+    slope = _slope(x_arr, y_arr)
+    intercept = y_mean - slope * x_mean
+    y_estim = slope * x_arr + intercept
+    slope_stderr = da.sqrt(((y_arr - y_estim)**2).sum() / dof /
+                           ((x_arr - x_mean)**2).sum())
+    return slope_stderr
+
+
+def _set_trend_units(cube, coord):
+    """Set correct trend units for cube."""
+    coord_units = coord.units
+    if coord_units.is_time_reference():
+        coord_units = Unit(coord_units.symbol.split()[0])
+    invalid_units = any([cube.units is None, cube.units.is_unknown(),
+                         cube.units.is_no_unit(), coord_units.is_no_unit()])
+    if not invalid_units:
+        cube.units /= coord_units
+
+
+def linear_trend(cube, coordinate='time'):
+    """Calculate linear trend of data along a given coordinate.
+
+    The linear trend is defined as the slope of an ordinary linear regression.
+
+    Parameters
+    ----------
+    cube : iris.cube.Cube
+        Input data.
+    coordinate : str, optional (default: 'time')
+        Dimensional coordinate over which the trend is calculated.
+
+    Returns
+    -------
+    iris.cube.Cube
+        Trends.
+
+    Raises
+    ------
+    iris.exceptions.CoordinateNotFoundError
+        The dimensional coordinate with the name ``coordinate`` is not found in
+        ``cube``.
+
+    """
+    coord = cube.coord(coordinate, dim_coords=True)
+    axis = cube.coord_dims(coord)[0]
+
+    # Calculate trend
+    trend_arr = da.apply_along_axis(
+        _get_slope, axis, cube.data, coord.points, dtype=cube.dtype,
+        shape=())
+    trend_arr = da.ma.masked_invalid(trend_arr)
+
+    # Create dummy cube by collapsing along dimension and add trend data
+    aggregator = iris.analysis.Aggregator('trend', _remove_axis_np,
+                                          lazy_func=_remove_axis_da)
+    cube = cube.collapsed(coord, aggregator)
+    cube.data = trend_arr
```

Comment:
Why are you not constructing the aggregator you actually need and then using that to compute the result?

Reply:
Good question -- I tried it first and saw that the `call_func` of [`iris.analysis.Aggregator`](https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.Aggregator) does only take the cube's `data` as argument, not the coordinate points which I need. Now I realize that I can probably provide the coordinate's points via `kwargs`. Let me check if that works.